crates/worker/src/partition_processor_manager.rs

```diff
@@ -501,8 +502,10 @@ impl<T: TransportConnect> PartitionProcessorManager<T> {
                 Some(command) = self.rx.recv() => {
                     self.on_command(command);
                 }
-                Some(update_processors) = self.incoming_update_processors.next() => {
-                    if let Err(err) = self.on_control_processors(update_processors).await {
+                Some(control_processors) = self.incoming_update_processors.next() => {
+                    let span = debug_span!(parent: None, "control_processors");
+                    span.follows_from(Span::current());
+                    if let Err(err) = self.on_control_processors(control_processors).instrument(span).await {
```

Comment:
This is so each `control_processors` message is handled in it's own span instead of having it a child of the processor loop which is really hard to view on `jaeger`

Reply:
The idea is without the extra span all `on_control_processors` spans will be a child of the partition processor manager loop. hence the span for that loop. It means ALL the `on_control_processors` will show up in one trace as with your screenshot.

With that extra span. the `on_control_processor` becomes a root trace (that links to the partition processor manager `run`) Hence it's easier to sort them by time, find a single event that happens at a certain time, etc...

Reply:
You can see that your view above will just get longer with each ControlProcessor message received by this node forever.